packages/components/src/components/tabs/shadow.tsx

```diff
@@ -42,53 +53,109 @@ export class KolTabs implements TabsAPI {
+		}
+
+		return offset;
+	};
+
+	private prevPossibleTabIndex = (tabs: TabButtonProps[], offset: number, step = 1): number => {
+		const nextOffset = offset - step;
+
+		if (nextOffset >= 0) {
+			if (tabs[nextOffset]._disabled) {
+				return this.prevPossibleTabIndex(tabs, offset, step + 1);
 			}
+			return nextOffset;
 		}
+
 		return offset;
 	};
 
 	private onKeyDown = (event: KeyboardEvent) => {
-		let selectedIndex: number | null = null;
-		switch (event.key) {
-			case 'ArrowRight':
-				selectedIndex = this.nextPossibleTabIndex(this.state._tabs, this.state._selected, 1);
+		switch (event.key as Key) {
+			case Key.ArrowRight:
+				this.goToNextTab(event);
 				break;
-			case 'ArrowLeft':
-				selectedIndex = this.nextPossibleTabIndex(this.state._tabs, this.state._selected, -1);
+			case Key.ArrowLeft:
+				this.goToPreviousTab(event);
+				break;
+			case Key.Space:
+			case Key.Enter:
+				this.activateFocusedTab(event);
 				break;
-		}
-		if (selectedIndex !== null) {
-			const tab = this.state._tabs[selectedIndex];
-			if (tab._on?.onSelect) {
-				tab._on?.onSelect(event, selectedIndex);
-			}
-			this.onSelect(event, selectedIndex);
 		}
 	};
 
-	private readonly onClickSelect = (event: MouseEvent, index: number): void => {
-		const tab = this.state._tabs[index];
-		if (tab._on?.onSelect) {
-			tab._on?.onSelect(event, index);
+	private getCurrentFocusIndex(): number {
+		if (typeof this.currentFocusIndex === 'number') {
+			return this.currentFocusIndex;
+		}
+
+		return this.state._selected;
+	}
+
+	private getKeyboardTabChangeMode(): 'selectFocusOnly' | 'activateComplete' {
+		if (this._behavior === 'select-manual') {
+			return 'selectFocusOnly';
 		}
-		this.onSelect(event, index);
+
+		return 'activateComplete';
+	}
+
+	private goToNextTab(event: KeyboardEvent) {
+		const nextFocusIndex = this.nextPossibleTabIndex(this.state._tabs, this.getCurrentFocusIndex());
+		this.selectNextTabEvent(event, nextFocusIndex, this.getKeyboardTabChangeMode());
+	}
+
+	private goToPreviousTab(event: KeyboardEvent) {
+		const nextFocusIndex = this.prevPossibleTabIndex(this.state._tabs, this.getCurrentFocusIndex());
+		this.selectNextTabEvent(event, nextFocusIndex, this.getKeyboardTabChangeMode());
+	}
+
+	private activateFocusedTab(event: KeyboardEvent) {
+		if (typeof this.currentFocusIndex === 'number') {
+			this.onSelect(event, this.currentFocusIndex);
+		}
+	}
+
+	private readonly onClickSelect = (event: MouseEvent, index: number): void => {
+		this.selectNextTabEvent(event, index);
 	};
 
+	private selectNextTabEvent(
+		event: KeyboardEvent | MouseEvent,
+		nextTabIndex: number,
+		changeMode: 'selectFocusOnly' | 'activateComplete' = 'activateComplete',
+	): void {
+		this.currentFocusIndex = nextTabIndex;
+
+		if (this.tabPanelsElement /* SSR instanceof HTMLElement */) {
+			const button: HTMLElement | null = koliBriQuerySelector(`button#${this.state._label.replace(/\s/g, '-')}-tab-${nextTabIndex}`, this.tabPanelsElement);
```

Comment:
Wollen wir das vielleicht in eine neue Methode packen (`focusTabById`)? Die Logik ist gerade dupliziert.

Reply:
ich hab hier nichts dupliziert, war vorher schon so, nur den change mode hinzugefügt

Reply:
@Makko74 bitte Zeile 450 beachten. Bitte den komplexen Selector in einen private Methode auslagern und wiederverwenden. Fokus bitte mitnehmen. -> `focusTabById`